src/script/common/30-composer.sh

```
@@ -0,0 +1,42 @@
+
+    log INFO "Installing Composer dependencies..."
+    cd /var/www/app
+
+    local -a cmd=(composer install --no-interaction --no-progress --optimize-autoloader)
+
+    if [[ "${YII_ENV:-}" == "prod" || "${BUILD_TYPE:-}" == "prod" ]]; then
+        cmd+=(--no-dev --classmap-authoritative)
+    fi
+
+    [[ "${COMPOSER_NO_SCRIPTS:-false}" == "true" ]] && cmd+=(--no-scripts)
+    [[ -n "${COMPOSER_EXTRA_FLAGS:-}" ]] && cmd+=(${COMPOSER_EXTRA_FLAGS})
```

Shellcheck (0.10.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)